bodhi/tests/client/test_bindings.py

```diff
@@ -24,6 +25,435 @@
+        """
+        client = bindings.BodhiClient()
+        client.send_request = mock.MagicMock(return_value='return_value')
+        client.csrf_token = 'a token'
+        kwargs = {
+            'builds': ['bodhi-2.4.0-1.fc26'], 'type_': 'enhancement', 'notes': 'This is a test',
+            'request': 'testing', 'autokarma': True, 'stable_karma': 3, 'unstable_karma': -3,
+            'severity': 'low'}
+
+        response = client.save(**kwargs)
+
+        self.assertEqual(response, 'return_value')
+        kwargs['type'] = kwargs['type_']
+        kwargs['csrf_token'] = 'a token'
+        client.send_request.assert_called_once_with('updates/', verb='POST', auth=True, data=kwargs)
+
+    def test_without_type_(self):
+        """
+        Assert correct operation when type_ isn't given.
+        """
+        client = bindings.BodhiClient()
+        client.send_request = mock.MagicMock(return_value='return_value')
+        client.csrf_token = 'a token'
+        kwargs = {
+            'builds': ['bodhi-2.4.0-1.fc26'], 'type': 'enhancement', 'notes': 'This is a test',
+            'request': 'testing', 'autokarma': True, 'stable_karma': 3, 'unstable_karma': -3,
+            'severity': 'low'}
+
+        response = client.save(**kwargs)
+
+        self.assertEqual(response, 'return_value')
+        kwargs['csrf_token'] = 'a token'
+        client.send_request.assert_called_once_with('updates/', verb='POST', auth=True, data=kwargs)
+
+
+class TestBodhiClient_save_override(unittest.TestCase):
+    """
+    This class contains tests for BodhiClient.save_override().
+    """
+    def test_save_override(self):
+        """
+        Test the save_override() method.
+        """
+        client = bindings.BodhiClient()
+        client.send_request = mock.MagicMock(return_value='return_value')
+        client.csrf_token = 'a token'
+        now = datetime.utcnow()
+
+        response = client.save_override('python-pyramid-1.5.6-3.el7', 2,
+                                        'This is needed to build bodhi-2.4.0.')
+
+        self.assertEqual(response, 'return_value')
+        actual_expiration = client.send_request.mock_calls[0][2]['data']['expiration_date']
+        client.send_request.assert_called_once_with(
+            'overrides/', verb='POST', auth=True,
+            data={'nvr': 'python-pyramid-1.5.6-3.el7',
+                  'expiration_date': actual_expiration,
+                  'notes': 'This is needed to build bodhi-2.4.0.', 'csrf_token': 'a token'})
+        # Since we can't mock utcnow() since it's a C extension, let's just make sure the expiration
+        # date sent is within 2 seconds of the now variable. It would be surprising if it took more
+        # than 2 seconds to start the function and execute its first instruction!
+        expected_expiration = now + timedelta(days=2)
+        self.assertTrue((actual_expiration - expected_expiration) < timedelta(seconds=2))
```

Comment:
This may be a flaky test. Not because it's likely it'll take more than two seconds to run (although I guess that's possibly with very unlucky scheduling), but because if a system clock update occurs between the first call and the second call to ``utcnow`` and the drift was more than 2 seconds (which seems like a very small number for drift) this won't be true.

Maybe that won't happen much (or ever!), just thought I'd bring it up.